test/normalCommand.test.ts

```diff
@@ -0,0 +1,74 @@
+import { newTest } from './testSimplifier';
+import { cleanUpWorkspace, setupWorkspace } from './testUtils';
+
+suite('Execute normal command', () => {
+  setup(async () => {
+    await setupWorkspace();
+  });
+
+  teardown(cleanUpWorkspace);
+
+  newTest({
+    title: 'One liner',
+    start: ['foo =| bar = 1'],
+    keysPressed: ':normal f=i!=\n',
+    end: ['foo = bar !|== 1'],
+  });
+
+  newTest({
+    title: 'One liner with selection',
+    start: ['foo =| bar = 1'],
+    keysPressed: 'V:normal f=i!=\n',
+    end: ['foo !|== bar = 1'],
+  });
+
+  newTest({
+    title: 'Multiple liner with selection',
+    start: ['foo =| bar = 1', 'foo = bar = 2'],
+    keysPressed: 'Vj:normal f=i!=\n',
+    end: ['foo !== bar = 1', 'foo !|== bar = 2'],
+  });
+
+  newTest({
+    title: 'Multiple liner with line range',
+    start: ['foo =| bar = 1', 'foo = bar = 2', 'foo = bar = 3'],
+    keysPressed: ':2,3normal f=i!=\n',
+    end: ['foo = bar = 1', 'foo !== bar = 2', 'foo !|== bar = 3'],
+  });
+
+  newTest({
+    title: 'One liner with dot',
+    start: ['foo =| bar = 1', 'foo = bar = 2'],
+    keysPressed: 'f=i!=<Esc>j^f=:normal .\n',
+    end: ['foo = bar !== 1', 'foo !|== bar = 2'],
+  });
+
+  // TODO: remove comment out after fixing dot command bug
+  // newTest({
+  //   title: 'One liner with multiple dot',
+  //   start: ['foo =| bar = 1', 'foo = bar = 2'],
+  //   keysPressed: 'f=i!=<Esc>j^f=:normal 2.\n',
+  //   end: ['foo = bar !== 1', 'foo !=!|== bar = 2'],
+  // });
+
+  newTest({
+    title: 'One liner with macro',
+    start: ['|1. one, 2. two, 3. three, 4. four'],
+    keysPressed: 'qaf.r)q:normal @a\n',
+    end: ['1) one, 2|) two, 3. three, 4. four'],
+  });
+
+  newTest({
+    title: 'One liner with multiple macro',
+    start: ['|1. one, 2. two, 3. three, 4. four'],
+    keysPressed: 'qaf.r)q:normal 3@a\n',
+    end: ['1) one, 2) two, 3) three, 4|) four'],
+  });
+
+  newTest({
+    title: 'Multiple liner with multiple macro',
+    start: ['|0. zero', '1. one, 2. two, 3. three, 4. four', '5. five, 6. six, 7. seven, 8. eight'],
+    keysPressed: 'qaf.r)qjVj:normal 4@a\n',
+    end: ['0) zero', '1) one, 2) two, 3) three, 4) four', '5) five, 6) six, 7) seven, 8|) eight'],
+  });
+});
```

Comment:
Add a test that shows `:normal` stops when a command fails. An easy way to do that is be near the end of a line and run `:normal lllllllllllllllllllj` - the `j` will never be executed because once you hit the EOL, `l` fails and you stop.

Reply:
I added commented out test on [853fb47](https://github.com/VSCodeVim/Vim/pull/8896/commits/853fb47879ab365b6b9e654c15213fbb0994706e).

Currently, operations specified throgh normal command is delegated to `modeHandler.handleMultipleKeyEvents` method, and we cannot detect operation failure    inside of the method.

I would like to fix it on another issue because it requires large change.

Reply:
Instead of commenting it out, please use `newTestSkip`

Reply:
Thanks, I modified test code.
[766ea77](https://github.com/VSCodeVim/Vim/pull/8896/commits/766ea772074c24642d5216cb9e6faf8f97652425)